### PR TITLE
Update/dependencies july2023

### LINKED ISSUE #765: update python version
Update python to `>=3.10`, and replace `setup.py` with `pyproject.toml`

- info: https://pip.pypa.io/en/stable/reference/build-system/pyproject-toml/
- rationale: https://stackoverflow.com/questions/62983756/what-is-pyproject-toml-file-for
- example: https://github.com/CSCfi/swift-browser-ui/blob/devel/pyproject.toml

### PR DESCRIPTION
**Pull request by blankdots**
### Description

Updating as we have a lot of dependencies security updates and tests failing

### Related issues

closes #765 

### Type of change

- [x] Update

### Changes Made

- update to python 3.11
- update python dependencies in one go
- update pre-commit packages
- fix pylint deprecation issues with distutil
- fix yaml syntax to be more uniform

### Testing

- [x] Tests do not apply

### Mentions

@gimppa we still need pyproject.toml so would be nice to have a PR with that, and maybe updating typing

### CODE REVIEW

File: docs/submitter.rst [hunk 1]
```diff
@@ -5,7 +5,7 @@ Metadata Submitter Backend
 
 .. note:: Requirements:
 
-  - Python 3.8+
+  - Python 3.10+
```
Comment (teemukataja) [inline docs/submitter.rst:8]:
```suggestion
  - Python 3.11+
```
[.pylintrc](https://github.com/CSCfi/metadata-submitter/blob/60f5397748a2ac62cd707a0a1b56f177c2430eaf/.pylintrc#L83-L85) and [pyproject.toml](https://github.com/CSCfi/metadata-submitter/blob/60f5397748a2ac62cd707a0a1b56f177c2430eaf/pyproject.toml#L10) have minimum version set as `3.11`, should this match that?

File: metadata_backend/api/handlers/submission.py [hunk 2]
```diff
@@ -53,7 +53,7 @@ async def get_submissions(self, req: Request) -> Response:
         if "published" in req.query:
             pub_param = req.query.get("published", "").title()
             if pub_param in {"True", "False"}:
-                submission_query["published"] = {"$eq": bool(strtobool(pub_param))}
+                submission_query["published"] = {"$eq": bool(str_to_bool(pub_param))}
```
Comment (teemukataja) [inline metadata_backend/api/handlers/submission.py:56]:
```suggestion
                submission_query["published"] = {"$eq": str_to_bool(pub_param)}
```
reduntant `bool()` (already boolean)

File: metadata_backend/conf/conf.py [hunk 3]
```diff
@@ -207,7 +226,7 @@ def create_db_client() -> AsyncIOMotorClient:
 DATACITE_SCHEMAS = {"study", "dataset", "bpdataset"}
 METAX_SCHEMAS = {"study", "dataset"}
 
-mq_ssl = bool(strtobool(os.environ.get("BROKER_SSL", "False")))
+mq_ssl = bool(str_to_bool(os.environ.get("BROKER_SSL", "False")))
```
Comment (teemukataja) [inline metadata_backend/conf/conf.py:229]:
```suggestion
mq_ssl = str_to_bool(os.environ.get("BROKER_SSL", "False"))
```
reduntant `bool()` (already boolean)

**Review by teemukataja — DISMISSED** [review]

**Review by teemukataja — APPROVED** [review]

**Review by csc-felipe — APPROVED** [review]

File: metadata_backend/api/handlers/submission.py [hunk 4]
```diff
@@ -194,7 +206,9 @@ async def get_submission(self, req: Request) -> Response:
 
         LOG.info("GET submission with ID: %r was successful.", submission_id)
         return web.Response(
-            body=ujson.dumps(submission, escape_forward_slashes=False), status=200, content_type="application/json"
+            body=ujson.dumps(submission, escape_forward_slashes=False),
+            status=200,
+            content_type="application/json",
```
Comment (csc-jm) [inline metadata_backend/api/handlers/submission.py:211]: Interesting that the new code style splits something like this to separate rows...

File: metadata_backend/helpers/parser.py [hunk 5]
```diff
@@ -9,15 +9,7 @@
...
-from xmlschema import (
-    ElementData,
-    XMLSchema,
-    XMLSchemaConverter,
-    XMLSchemaException,
-    XsdElement,
-    XsdType,
-    aliases,
-)
+from xmlschema import ElementData, XMLSchema, XMLSchemaConverter, XMLSchemaException, XsdElement, XsdType, aliases
```
Comment (csc-jm) [inline metadata_backend/helpers/parser.py:12]: ...while also reverting these multi-line imports back to a single line.

**Review by csc-jm — APPROVED** [review]
I find the code style update rather intriguing but I don't have any problems with them, so approving

**Pull request merged**